cpp/include/cudf/search.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cudf.h"
+#include "types.hpp"
+#include <vector>
+
+namespace cudf {
+
+/**---------------------------------------------------------------------------*
+ * @brief Find smallest indices in sorted column where values should be 
+ *  inserted to maintain order
+ * 
+ * @param column        Column to search in
+ * @param values        Values to search insert locations for
+ * @param descending    Indicate if the column is sorted in descending order.
+ *  If false, the column is considered sorted ascending.
+ * @param nulls_as_largest If true, nulls are considered larger than valid
+ *  values, otherwise, nulls are considered smaller than valid values
+ * 
+ * @return gdf_column   Insertion points with the same shape as values
+ *  Non-nullable column of type GDF_INT32
```

Comment:
```suggestion
```

Reply:
@jrhemstad asked for it https://github.com/rapidsai/cudf/pull/2156#discussion_r302059688

Reply:
@harrism just moved it up to the line above.

Reply:
oops